curriculum/challenges/english/blocks/lecture-working-with-arrays-variables-and-naming-practices/6732c6ba2ea42b610b9f9ce1.md

```diff
@@ -20,57 +20,41 @@ console.log(fruits.length); // 3
 
 It's possible to have arrays with empty slots. Empty slots are defined as slots with nothing in them. This is different than an array with the value of `undefined`. These types of arrays are known as sparse arrays. Here is an example:
 
-:::interactive_editor
-
 ```js
 const sparseArray = [1, , , 4];
 console.log(sparseArray.length); // 4
 ```
 
-:::
-
-In this case even though we only have two defined elements, `1` and `4`. The length is `4` because the highest index (`3`) plus `1` gives us a length of `4`.
+In this case even though we only have two defined elements, `1` and `4`, the length is `4` because the highest index (`3`) plus `1` gives us a length of `4`. The empty slots in the middle do not contain `undefined`; they are truly empty.
 
 Now let's discuss how to create an empty array of fixed length. There are few ways to do this in JavaScript but one common method is to use the `Array()` constructor with a numeric argument. The `Array()` constructor can be used with the `new` keyword to create a new array. Here is an example:
 
-:::interactive_editor
-
 ```js
 const emptyArray = new Array(5);
 console.log(emptyArray.length); // 5
-console.log(emptyArray); // [undefined, undefined, undefined, undefined, undefined]
+console.log(emptyArray); // [ , , , , ]
 ```
 
-:::
-
-In this example, we create a new array using the `Array(5)`. This creates a sparse array with a length of `5` where all the slots are empty. 
+In this example, we create a new array using `Array(5)`. This creates a sparse array with a length of `5` where all the slots are empty (not containing `undefined`, but truly empty). Note: different environments may display empty slots differently, but they all represent the same sparse structure. 
```

Comment:
same issue here. Mentioning undefined here will be confusing because the code example above doesn't show that.

```suggestion
In this example, we create a new array using `Array(5)`. This creates a sparse array with a length of `5` where all the slots are empty. 
```